### PR TITLE
add coverage actions

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #811

### CODE REVIEW

**Comment by zjb0807** [comment]
Need add `codecov/codecov-action@v1` and `actions/cache@v2`

File: .github/workflows/coverage.yml
Comment (xlc) [inline .github/workflows/coverage.yml:6]: we want to run code coverage for every pr & master, just like unit tests

Comment (zjb0807) [reply .github/workflows/coverage.yml:6]: This needs to use self-host.  Running the code coverage for more than 1 hour.

Comment (xlc) [reply .github/workflows/coverage.yml]: ok. I can setup another self-host server

Comment (xlc) [reply .github/workflows/coverage.yml]: also we don't need to wait for coverage result before merge PR

**Review by xlc — APPROVED** [review]

**Pull request merged**
